src/rpc/fam_rpc_service_impl.cpp

```diff
@@ -240,8 +244,18 @@ Fam_Rpc_Service_Impl::destroy_region(::grpc::ServerContext *context,
         response->set_errormsg(e.fam_error_msg());
         return ::grpc::Status::OK;
     }
-    RPC_SERVICE_PROFILE_END_OPS(destroy_region);
 
+    int ret = deregister_region_memory(request->regionid());
+
+    if (ret < 0) {
```

Comment:
Here we should return back only if its permission/user related error and for memory server related error should be handled internally in memory server.

Reply:
deregister_region_memory will not return with error for failure to unregister dataitems. Anyway region is successfully destroyed by the nvmm allocator.

Reply:
we should not free the region ID if this happens and free it during app exit